src/rdkafka_queue.c

```diff
@@ -548,8 +549,8 @@ static size_t rd_kafka_purge_outdated_messages(int32_t version,
         for (i = 0; i < cnt; i++) {
                 rd_kafka_op_t *rko;
                 rko = rkmessages[i]->_private;
-                if (rd_kafka_op_version_outdated(rko, version)) {
-                        /* This also destroys the corresponding rkmessage. */
+                if (rko->rko_rktp == rktp &&
+                    rd_kafka_op_version_outdated(rko, version)) {
```

Comment:
Looking at op_version_outdated(), I think we might want to call it version set to 0 and let it just fall down to its 3rd if-statement:
https://github.com/edenhill/librdkafka/blob/master/src/rdkafka_partition.h#L806